include/circt/Dialect/SV/SVStatements.td

```diff
@@ -527,6 +527,20 @@ def CoverOp : SVOp<"cover", [ProceduralOp]> {
     "custom<OmitEmptyStringAttr>($label) attr-dict $property `:` type($property)";
 }
 
+def CoverConcurrentOp : SVOp<"cover.concurrent", []> {
+  let summary = "concurrent cover statement, i.e., cover property";
+  let description = [{
+    Assert that a specific property happens during the course of execution.
+  }];
+
+  let arguments =
+    (ins EventControlAttr:$event, I1:$clock, AnyType:$property, StrAttr:$label);
+  let results = (outs);
+  let assemblyFormat =
+    "custom<OmitEmptyStringAttr>($label) $event $clock $property attr-dict `:`"
+    " type($property)";
```

Comment:
What types are $property allowed to have?  Would it be reasonable to constrain it to `i1` unconditionally for now?  If there are usecases for other types, could you add a test for it?

Reply:
Technically, this is closer to a naive extension of the existing immediate cover that we have than full property support. For that reason, this should be `i1`. It turns out I had cargo-culted the `AnyType` from my original implementation of immediate assert/assume/cover. 😂 I'll change those to `i1` and update this here.

Eventually, we'll need something closer to an actual property in the IR (which is a very beefy thing). This concurrent cover, for now, is just a stop-gap to get property-style assertions to come out of the emitter. (Which I think is justified.)

Reply:
Change this to `I1` on `main` for the existing immediate assertion ops. Then changed here.